src/CloudNative.CloudEvents.NewtonsoftJson/JsonEventFormatter.cs

```diff
@@ -136,6 +136,44 @@ public override CloudEvent DecodeStructuredModeMessage(Stream data, ContentType
         public override CloudEvent DecodeStructuredModeMessage(byte[] data, ContentType contentType, IEnumerable<CloudEventAttribute> extensionAttributes) =>
             DecodeStructuredModeMessage(new MemoryStream(data), contentType, extensionAttributes);
 
+        public override async Task<IReadOnlyList<CloudEvent>> DecodeBatchModeMessageAsync(Stream data, ContentType contentType, IEnumerable<CloudEventAttribute> extensionAttributes)
+        {
+            data = data ?? throw new ArgumentNullException(nameof(data));
+
+            var jsonReader = CreateJsonReader(data, contentType.GetEncoding());
+            var array = await JArray.LoadAsync(jsonReader).ConfigureAwait(false);
+            return DecodeJArray(array, extensionAttributes);
+        }
+
+        public override IReadOnlyList<CloudEvent> DecodeBatchModeMessage(Stream data, ContentType contentType, IEnumerable<CloudEventAttribute> extensionAttributes)
```

Comment:
Do we know if there will ever be a Binary batch message as part of the spec?

Reply:
I've seen no hint of that, no. I can't see how it would work to be honest, given that headers are used for the attributes, and each event would need its own set of attributes. (I'm sure it could be done, but I suspect it would be massively clunky compared with just using the current approach.) I think we're okay to design the SDK (and method names in particular) assuming that batch always means "in a structured mode style".